### PR TITLE
[WIP] Add RouterOS API support to routeros_command module

### PR DESCRIPTION
**Pull request by heuels**
##### SUMMARY

This pull request adds support for issuing commands via RouterOS API and `local` connection type. This is required because of several problems with `network_cli` implementation:
- The output of the command gets wrapped by the RouterOS console and it sometimes makes the command itself appear in the output.
- The output is dependent on the terminal width. If one tries to set terminal width to a very large number, each line of the output will be equally distributed among the available width.
- Some commands with verbose output sometimes do not return values (like `/ip firewall filter print`).

Why use a **provider**? Because RouterOS has a weird and non-standard API. The choice was either to use `connection: local` and a **provider** or create a new module solely for RouterOS API support. After a discussion with **ganeshrn** [in the original `routeros_command` PR](https://github.com/ansible/ansible/pull/41155#discussion_r205900935) it was decided it would be better to use the former.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- `lib/ansible/module_utils/network/routeros/routeros.py`
- `lib/ansible/modules/network/routeros/routeros_command.py`

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (routeros_api 790c0bf2b3) last updated 2018/08/26 11:31:57 (GMT +300)
  config file = None
  configured module search path = ['/Users/heuels/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/heuels/GitHub/ansible/lib/ansible
  executable location = /Users/heuels/GitHub/ansible/bin/ansible
  python version = 3.6.5 (default, Jun 19 2018, 01:03:38) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by felixfontein** [comment]
@heuels Cool! I'll try it out later; I'm looking forward to be able to use this one as well! ;)

**Comment by heuels** [comment]
@felixfontein I'm looking forward to it! I still need to verify that SSL connection works but the basic stuff is ready :)

**Comment by shaders** [comment]
Please see: https://github.com/ansible/ansible/commit/663cdb5c8ca88ead4c949bcb32d34e3505dffe60#r30480308

**Comment by heuels** [comment]
@shaders, thank you! I've addressed your comment in b79c04a.

File: lib/ansible/module_utils/network/routeros/routeros.py [hunk 1]
```diff
@@ -110,47 +122,195 @@ def get_config(module, flags=None):
...
+    out = connection.edit_config(commands)
+
 
-    for cmd in to_list(commands):
-        if isinstance(cmd, dict):
-            command = cmd['command']
-            prompt = cmd['prompt']
-            answer = cmd['answer']
+def is_api(module):
+    display.debug(module.params)
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/routeros/routeros.py:136]: `display` won't work in module code, hence this line should be removed

File: lib/ansible/module_utils/network/routeros/routeros.py [hunk 2]
```diff
@@ -25,50 +25,62 @@
...
-from ansible.module_utils.connection import Connection
+from ansible.module_utils.connection import Connection, ConnectionError
 
 _DEVICE_CONFIGS = {}
 
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/routeros/routeros.py:45]: `display` should not be used in the module code. Please remove this import

File: lib/ansible/module_utils/network/routeros/routeros.py [hunk 3]
```diff
@@ -25,50 +25,62 @@
...
+
+try:
+    import librouteros
+    HAS_LIBROUTEROS = True
+except ImportError as e:
+    display.debug('IMPORT ERROR: %s' % e)
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/routeros/routeros.py:51]: Remove this line.

**Comment by abenokraitis** [comment]
@heuels Still waiting on your review of the comments from the Ansible team.

**Comment by abenokraitis** [comment]
needs_info

**Comment by heuels** [comment]
@abenokraitis Thanks for the reminder! I will find the time and address the comments this week.

**Comment by felixfontein** [comment]
@heuels any progress? Do you need help with something (like testing)?

BTW, if this should end in Ansible 2.8, it needs to be merged by [March 21st](https://docs.ansible.com/ansible/devel/roadmap/ROADMAP_2_8.html#expected).

**Comment by heuels** [comment]
I'm finally going to go ahead and close it since I have not managed to found any real uses for the API connection as of yet.

**Pull request closed without merging**